zorg/buildbot/commands/NinjaCommand.py

```diff
@@ -1,110 +1,86 @@
-# TODO: Use Interpolate instead of WithProperties.
 import re
 
-from buildbot.process.properties import WithProperties
-from buildbot.steps.shell import WarningCountingShellCommand
+from twisted.internet import defer
+from twisted.python import log as logging
 
-class NinjaCommand(WarningCountingShellCommand):
-    DEFAULT_NINJA = 'ninja'
+from buildbot.plugins import steps, util
 
-    @staticmethod
-    def sanitize_kwargs(kwargs):
-        # kwargs we could get and must not pass through
-        # to the buildstep.RemoteShellCommand constructor.
-        # Note: This is a workaround of the buildbot design issue,
-        # thus should be removed once the original issue gets fixed.
-        consume_kwargs = [
-                             "jobs",
-                             "loadaverage",
-                         ]
 
-        sanitized_kwargs = kwargs.copy()
-        for k in consume_kwargs:
-            if k in sanitized_kwargs.keys():
-                del sanitized_kwargs[k]
-
-        return sanitized_kwargs
+class NinjaCommand(steps.WarningCountingShellCommand):
+    DEFAULT_NINJA = "ninja"
 
     name = "build"
     haltOnFailure = True
     description = ["building"]
     descriptionDone = ["build"]
-    renderables = (
-        'options',
-        'targets',
-        'ninja',
-    )
 
-    def __init__(self, options=None, targets=None, ninja=DEFAULT_NINJA, logObserver=None, **kwargs):
+    renderables = ["options", "targets", "ninja", "jobs", "loadaverage"]
+
+    def __init__(
+        self,
+        options=None,
+        targets=None,
+        ninja=DEFAULT_NINJA,
+        logObserver=None,
+        jobs=None,
+        loadaverage=None,
+        **kwargs,
+    ):
         self.ninja = ninja
         self.targets = targets
+        self.options = options or []
 
-        if options is None:
-            self.options = list()
-        else:
-            self.options = list(options)
+        # The options are always list().
```

Comment:
Not sure this is true. It could be a string, as handled right below, or it could be a generator or a callable.

Reply:
we always convert an argument into a list. See few lines below.

Reply:
I have updated the code. The `options` must be iterable on None. It is used here
```
if self.options:
    ninja_command.extend(self.options)
```
Currently `options` is specified only in OpenMPBuilder.py as `options=['-k0']`. 
So we can drop the string format for the options.

Reply:
Sounds good.